core/conversion/evaluators/aten.cpp

```diff
@@ -172,6 +172,52 @@ auto aten_registrations TORCHTRT_UNUSED =
                auto out_tensor = torch::full(args.at(n->input(0)).unwrapToIntList().vec(), scalar_value, options);
                return out_tensor;
              }})
+        .evaluator(
+            {c10::Symbol::fromQualString("aten::full_like"),
+             // aten::full_like(Tensor self, Scalar fill_value, *, ScalarType? dtype=None, Layout? layout=None,
+             // Device? device=None, bool? pin_memory=None, MemoryFormat? memory_format=None) -> (Tensor)
+             [](const torch::jit::Node* n, kwargs& args) -> c10::optional<torch::jit::IValue> {
+               // Override options related to layout and device for TensorRT
+               auto options = torch::TensorOptions().layout(torch::kStrided).device(torch::kCUDA);
+               auto input_tensor_var = args.at(n->input(0));
+
+               std::vector<int64_t> input_shape;
+               c10::ScalarType input_dtype;
+
+               // Extract default data type and shape of input tensor, as required by full_like
+               if (input_tensor_var.isITensor()) {
+                 auto tensor = input_tensor_var.ITensor();
+                 input_shape = util::toVec(tensor->getDimensions());
+                 input_dtype = util::TRTDataTypeToScalarType(tensor->getType());
+               } else if (input_tensor_var.IValue()->isTensor()) {
+                 auto tensor = input_tensor_var.unwrapToTensor();
+                 input_shape = tensor.sizes().vec();
+                 input_dtype = tensor.scalar_type();
+               } else if (input_tensor_var.IValue()->isCustomClass()) {
+                 auto tensor = input_tensor_var.IValue()->toCustomClass<TensorContainer>()->tensor();
+                 input_shape = util::toVec(tensor->getDimensions());
+                 input_dtype = util::TRTDataTypeToScalarType(tensor->getType());
+               } else {
+                 TORCHTRT_THROW_ERROR(
+                     "IValue is not some class of Tensor. Found: " << input_tensor_var.IValue()->type());
```

Comment:
Invalid IValue type. IValue is not some class of torch::Tensor or nvinfer1::ITensor. Found .....